x/mongo/driver/compression.go

```diff
@@ -25,6 +26,21 @@ type CompressionOpts struct {
 	UncompressedSize int32
 }
 
+var zstdEncoders = &sync.Map{}
```

Comment:
Haven't seen a `sync.Map` in our production code before! I think it's normally used instead of a regular map and a mutex for the following reasons:

> The Map type is optimized for two common use cases: (1) when the entry for a given key is only ever written once but read many times, as in caches that only grow, or (2) when multiple goroutines read, write, and overwrite entries for disjoint sets of keys. In these two cases, use of a Map may significantly reduce lock contention compared to a Go map paired with a separate Mutex or RWMutex.

I think (1) is true in this case, as we seem to create an encoder for each encoder level and then "read" it (`Load`) many times.